### PR TITLE
Modify CMake build system to also support MSVC.

### PR DESCRIPTION
**Pull request by waddlesplash**
Essentially just the build system portion of #2946 with some MSVC-centric changes.

Does not build because the PCH-less build on Windows is busted, and the old MSVC PCH CMake stuff was busted and I wasn't able to fix it.

### CODE REVIEW

**Comment by waddlesplash** [comment]
cc @lioncash 
also @rohit-n, if you want to attempt to fix the no-PCH Windows build after this is merged, feel free.

File: CMakeLists.txt [hunk 1]
```diff
@@ -158,20 +130,24 @@ else()
...
+	   ${LOWERCASE_SYSTEM_PROCESSOR} MATCHES "i.86" OR
+	   ${LOWERCASE_SYSTEM_PROCESSOR} MATCHES "amd64" OR
 	   APPLE)
 		if(_ARCH_64)
 			set(_M_X86 1)
 			set(_M_X86_64 1)
-			add_definitions(-D_M_X86=1 -D_M_X86_64=1 -msse2)
+			add_definitions(-D_M_X86=1 -D_M_X86_64=1)
+			if(NOT MSVC)
+				add_definitions(-msse2)
```
Comment (phire) [inline CMakeLists.txt]: Do we need -msse2 anymore?

It's implied by X86_64

File: Source/Core/Common/Common.h [hunk 2]
```diff
@@ -82,10 +82,12 @@ class NonCopyable
 	#define GC_ALIGNED64_DECL(x) __declspec(align(64)) x
 
 // Since they are always around on Windows
+#if !defined HAVE_WX || !defined HAVE_OPENAL || !defined HAVE_PORTAUDIO
```
Comment (phire) [inline Source/Core/Common/Common.h]: Do it properly please.

**Comment by phire** [comment]
At a glance it looks OK, but I won't be happy merging until it works.

**Comment by shuffle2** [comment]
dude, fucking work on qt

**Comment by delroth** [comment]
Exact same thing as the other PR you reference:

Do we want to support CMake/MSVC builds, and if so why? What is your plan to deprecate the non-CMake MSVC builds, because we clearly don't want two solutions that do the same thing.

I veto this PR being merged (and will revert any merge) unless these questions have a satisfactory answer (satisfying either me or @shuffle2 — you may choose whoever you think might be easiest to convince).

File: CMakeLists.txt (hunk 1 above)
Comment (waddlesplash) [inline CMakeLists.txt]: Probably not. Will remove.

**Comment by waddlesplash** [comment]
> dude, fucking work on qt

I *really* don't like how VS integrates with Qt, so I was at least going to try to use Qt Creator for development on Windows. The overhead to have MSVC builds work on the CMake build system is not very high. So if at all possible, I really would prefer that this is merged so I can code in Qt Creator instead of MSVC on Windows.

**Comment by waddlesplash** [comment]
... Whatever. The amount of time and effort it would take to convince you is probably just not worth it on my part.

**Comment by phire** [comment]
I would support windows cmake support if it could generate build files that produce a comparable result to the current hand-written build (both in the build result, and the developer-experience when using visual studio)

It is a little annoying to have to update both CMakeLists.txt and the .vcxproj xml files, especially on windows systems. My Stipulations are: it must be done with the intention that we "probably will" replace the hand-written files with cmake generated files later and if we are supporting both, there must be buildbots which make use of both to error when someone messes up.

**Pull request closed without merging**